facefusion/audio.py

```diff
@@ -0,0 +1,64 @@
+from typing import Optional, Any, List
+
+import numpy
+import scipy
+from functools import lru_cache
+from facefusion.ffmpeg import read_audio_buffer
+from facefusion.typing import Fps, Audio, Spectrogram, AudioFrame
+
+
+def get_audio_frame(audio_path : str, fps : Fps, frame_number : int) -> Optional[AudioFrame]:
+	if audio_path:
+		audio_frames = read_static_audio(audio_path, fps)
+		if frame_number < len(audio_frames):
+			return audio_frames[frame_number]
+	return None
+
+
+@lru_cache(maxsize = None)
+def read_static_audio(audio_path : str, fps : Fps) -> List[Optional[AudioFrame]]:
+	audio_buffer = read_audio_buffer(audio_path, 16000, 2)
+	audio = numpy.frombuffer(audio_buffer, dtype = numpy.int16).reshape(-1, 2)
+	audio = normalize_audio(audio)
+	spectrogram = create_spectrogram(audio, 16000, 80, 800)
+	audio_frames = extract_audio_frames(spectrogram, 80, 16, fps)
+	return audio_frames
+
+
+def normalize_audio(audio : numpy.ndarray[Any, Any]) -> Audio:
+	if audio.ndim > 1:
+		audio = numpy.mean(audio, axis = 1)
+	audio = audio / numpy.max(numpy.abs(audio), axis = 0)
+	audio = scipy.signal.resample(audio, len(audio))
+	audio = scipy.signal.lfilter([1, -0.97], [1], audio)
+	return audio
+
+
+@lru_cache(maxsize = None)
+def create_static_mel_filter(sample_rate : int, filter_total : int, filter_size : int) -> numpy.ndarray[Any, Any]:
+	bins = (10 ** (numpy.linspace(85, 2787, filter_total + 2) / 2595) - 1) * 700
```

Comment:
we cannot leave it that way - this are kinda magic numbers that only you understand.

but neither me or your future self do :D

Reply:
i managed to move those numbers inside `read_static_audio()`.
what do you suggest for storing the values? A global dict like this?
```python
AudioParameters : Dict[Any] = {
    ''sample_rate": 16000,
    "frequency_minimum": 55.0,
    "frequency_maximum": 7600.0,
    "filter_total": 80,
    "filter_size": 800,
    "filter_coefficient: -0.97"
}
```